File: DashboardDemo/DashboardSampleVB/Dashboard/TaskElement.vb
Imports Telerik.WinControls.UI

Public Class TaskElement
    Inherits LightVisualElement

    Public Sub New(ByVal text As String)
        Me.Text = text
    End Sub

    Protected Overrides Sub InitializeFields()
        MyBase.InitializeFields()
        Me.DrawFill = True
        Me.DrawBorder = True
        Me.GradientStyle = Telerik.WinControls.GradientStyles.Solid
        Me.BorderGradientStyle = Telerik.WinControls.GradientStyles.Solid
        Me.BackColor = Color.Yellow
        Me.BorderColor = Color.Black
        Me.AllowDrag = True
        Me.StretchHorizontally = True
        Me.StretchVertically = False
        Me.TextWrap = True
        Me.Padding = New System.Windows.Forms.Padding(10)
        Me.Margin = New System.Windows.Forms.Padding(5)
    End Sub

    Protected Overrides Sub OnMouseDown(ByVal e As System.Windows.Forms.MouseEventArgs)
        MyBase.OnMouseDown(e)
        Dim dashboard As DashboardElement = FindAncestor(Of DashboardElement)()
        dashboard.DragDropService.Start(Me)
    End Sub
End Class

File: DashboardDemo/DashboardSample/Dashboard/TaskElement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Telerik.WinControls.UI;
using System.Drawing;

namespace TicketTest
{
    public class TaskElement: LightVisualElement
    {
        public TaskElement(string text)
        {
            this.Text = text;
        }

        protected override void InitializeFields()
        {
            base.InitializeFields();

            this.DrawFill = true;
            this.DrawBorder = true;
            this.GradientStyle = Telerik.WinControls.GradientStyles.Solid;
            this.BorderGradientStyle = Telerik.WinControls.GradientStyles.Solid;
            this.BackColor = Color.Yellow;
            this.BorderColor = Color.Black;
            this.AllowDrag = true;
            this.StretchHorizontally = true;
            this.StretchVertically = false;
            this.TextWrap = true;
            this.Padding = new System.Windows.Forms.Padding(10);
            this.Margin = new System.Windows.Forms.Padding(5);
        }

        protected override void OnMouseDown(System.Windows.Forms.MouseEventArgs e)
        {
            base.OnMouseDown(e);
            
            DashboardElement dashboard = FindAncestor<DashboardElement>();
            dashboard.DragDropService.Start(this);
        }
    }
}
